### PR TITLE
chore(modal): rename to dds-expressive-modal

### PR DESCRIPTION
**Pull request by asudoh**
### Related Ticket(s)

Refs #4391.

### Description

Renames `<dds-modal>` to `<dds-expressive-modal>` upon our designer's feedback. All `<dds-modal-*>` components are also renamed as such.

Also:

* Renames `DDSModal` class name to `DDSExpressiveModal`.
* Renames `expressive-size` attribute to `size`.
* Renames `DDS_MODAL_SIZE` enum to `EXPRESSIVE_MODAL_SIZE`.

### Changelog

**Changed**

- Renames `<dds-modal>` to `<dds-expressive-modal>`
- Renames `<dds-modal-*>` components to `<dds-expressive-modal-*>`.
- Renames `DDSModal` class name to `DDSExpressiveModal`.
- Renames `expressive-size` attribute to `size`.
- Renames `DDS_MODAL_SIZE` enum to `EXPRESSIVE_MODAL_SIZE`.

### CODE REVIEW

**Comment by jeffchew** [comment]
Thank you for this @asudoh , looks good!

Can you update the README, it still references as `dds-modal`.

**Comment by asudoh** [comment]
> Can you update the README, it still references as `dds-modal`.

Good catch @jeffchew - Updated.

File: packages/web-components/examples/codesandbox/components/modal/index.html [hunk 1]
```diff
@@ -27,19 +27,19 @@ <h1>Hello World! 👋</h1>
     <div id="app">
```
Comment (jeffchew) [inline packages/web-components/examples/codesandbox/components/modal/index.html:27]: @asudoh I think the folder name should change too to be consistent.

File: packages/web-components/src/components/expressive-modal/__stories__/README.stories.mdx [hunk 2]
```diff
@@ -45,19 +45,19 @@ or with [JSPM](https://jspm.org)
 ### HTML
 
 ```html
-<dds-modal open>
-  <dds-modal-header>
-    <dds-modal-close-button role="button"></dds-modal-close-button>
-    <dds-modal-heading>Modal Title</dds-modal-heading>
-  </dds-modal-header>
-  <dds-modal-body>
+<dds-expressive-modal open>
```
Comment (jeffchew) [inline packages/web-components/src/components/expressive-modal/__stories__/README.stories.mdx:48]: The JS documentation above I think also needs to change right? the filenames and folders are still the old ones.

File: packages/web-components/examples/codesandbox/components/modal/index.html (hunk 1 above)
Comment (asudoh) [reply packages/web-components/examples/codesandbox/components/modal/index.html:27]: Great catch @jeffchew - Fixed.

File: packages/web-components/src/components/expressive-modal/__stories__/README.stories.mdx (hunk 2 above)
Comment (asudoh) [reply packages/web-components/src/components/expressive-modal/__stories__/README.stories.mdx:48]: Great catch @jeffchew - Fixed.

**Review by jeffchew — APPROVED** [review]

**Comment by jeffchew** [comment]
> This PR currently has a merge conflict. Please resolve this and then re-add the `Ready to merge` label.

@asudoh can you resolve the merge conflicts?

**Comment by asudoh** [comment]
@jeffchew Done!

**Pull request merged**